src/docs/commands_keybindings.md

```diff
@@ -1,122 +1,138 @@
 ---
-title: Commands and Keybindings
+title: Commands/Menus/Keybindings
 ---
 
-# Commands and Keybindings
+# Commands, Menus and Keybindings
 
-Theia can be extended in a number of different ways. Commands allow packages to contribute unique commands that can then be called by other packages. It's also possible to add keybindings and contexts to these commands, so that they can only be called under certain conditions (window focus, current selection etc.).
+Commands are runnable actions defined by an ID, a name and the function to be executed. Commands can be triggered via the quick access bar, they can be bound to keybindings or menu items and they can be called programmatically. Commands can be context sensitive, so that they can only be called under certain conditions (window focus, current selection etc.).
 
-## Adding commands to Theia command registry
+The following sections provide details about how to contribute commands, keybindings and menu items, how to connect them and how to use the corresponding services for managing these items.
 
-To contribute commands to the command registry, modules must implement the `CommandContribution` class, i.e
+If you are not yet familiar with contribution points in Theia or the use of dependency injection, please consider this guide on [Services and Contributions](https://theia-ide.org/docs/services_and_contributions/).
 
-**java-commands.ts**
+All the following code examples are from the [Theia extension generator](https://github.com/eclipse-theia/generator-theia-extension). You can get the same code set-up by installing the generator, selecting the “Hello World” example (see here) and choosing “helloworld” as the name. 
+
+## Contributing commands
+
+By contributing a command, you can add a custom action to Theia. A command can be triggered by the user via the quick access bar, a key binding or a menu entry. It can also be called programmatically.
+
+All commands of a Theia application are managed in the `CommandRegistry`. To contribute commands to the command registry, modules must implement the `CommandContribution` interface (see code example below). 
+
+A command is an object with an `id` and an optional, potentially user visible `label` (see HelloworldCommand in the example below). The command contribution receives the `CommandRegistry` in the `registerCommands` function as a parameter. The command can then be registered by calling `registerCommand` on this registry. Along with the command  a callback needs to be provided which is executed whenever the command is triggered. In the example, the command uses the MessageService to “say hello”. 
+
+**helloworld-contribution.ts**
 ```typescript
+export const HelloworldCommand = {
```

Comment:
We should give users `typings` here, if not done by the generator already we should add it:

```suggestion
export const HelloworldCommand: Command = {
```